Bugfixes and feature updates

carlosingles opened the pull request:
This PR contains fixes for two bugs:

- When opening files that are already in the "Open Files" list, they would double up
- When browsing FTP directories, the current and parent directory would double up

And a few feature upgrades that I felt were necessary:

- When the package initialises, the "Open Files" list will get cleared
- When browsing a host, the current directory will get saved, returning you to that directory when you browse the host again

I'm not sure if the features I added should be default behaviour, or if they should be added in as a Setting to let the user decide the behaviour.

sveale commented:
This looks great! Good work :)
I think it would be beneficial if the two new features were disabled as default and toggleable in settings.
Do you think you could push that as well? That way I can just accept the PR

carlosingles commented:
Updated again, previous features now have Settings that control their behaviour

I have also added a new setting which allows you to filter folders on top when browsing a directory, as well as a path indicator at the top of the modal. 
**Note: The current path is being placed in the error message div, this may not be ideal**

I also removed the current directory entry that would appear, I felt that this is unnecessary since it doesn't really do anything.

Finally, I noticed that the modals that were being created were not being removed from the document after they had been dismissed, so I fixed this by calling destroy() on the panel when it is being destroyed

sveale commented:
Great work. Thanks for the contribution :). Will push new version to atom.io shortly.
